js/src/common/states/PaginatedListState.ts

```diff
@@ -0,0 +1,171 @@
+import * as Mithril from 'mithril';
+
+import Model from '../Model';
+
+export interface Page<TModel> {
+  number: number;
+  items: TModel[];
+
+  // // maybe we can put page links here?
+  // links: {[name: string]: string},
+  // // or
+  // previousPage?: string;
+  // nextPage?: string;
+}
+
+export interface PaginationLocation {
+  page: number;
+  startIndex?: number;
+  endIndex?: number;
+}
+
+export default abstract class PaginatedListState<T extends Model> {
+  // TODO:
+  //  - what does this page track? last page request? last page that is loaded
+  //      (e.g. pg 7 if u have 3-7 loaded)
+  //  - When do we set it?
+  protected location!: PaginationLocation;
```

Comment:
`pages` is a contiguous structure that expands from both ends. So it could either be an array or a doubly linked list. If it's a doubly linked list, `location`'s `page` prop should point to the active element. If it's an array, `location`'s `page` prop should be the index of the active page (although we'd need to remember to increment by 1 when adding new pages.

Reply:
What are we defining with "active" page though? If the user loads the previous page, does the active page still remain the last number in ascending order (e.g. loaded pg 2 but pg 5 is still active) or does it change to page 2?

Reply:
Would the previous page be the page that the user is currently on?

Reply:
What are you referring to with previous page? I meant as load next page, load prev page, for loading a page that hasn't been loaded.

Reply:
Active page, sorry.